test/events.js

```diff
@@ -22,6 +22,44 @@ describe('events', function () {
     })
   })
 
+  it('waits on connect until finished', function (done) {
+    var pool = new Pool()
+    var emittedClient = false
+    pool.on('connect', function (client) {
+      return client.query('SELECT now()')
```

Comment:
why do you need to actually call query? looks irrelevant for this test.

Reply:
i want an async function to be called

Reply:
then go to sleep 😴 . calling query will trigger new sequence of event being emitted that might affect results of this test.

Reply:
sleeping now :)